src/components/editor/SinopiaResourceTemplates.jsx

```diff
@@ -0,0 +1,115 @@
+// Copyright 2019 Stanford University see Apache2.txt for license
+
+import React, {Component} from 'react'
+import BootstrapTable from 'react-bootstrap-table-next'
+const SinopiaServer = require('sinopia_server')
+const instance = new SinopiaServer.LDPApi()
+instance.apiClient.basePath = 'http://localhost:8080'
+
+class SinopiaResourceTemplates extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      isLoading: false,
+      groupData: [],
+      templatesForGroup: []
+    }
+  }
+
+  async componentDidMount() {
```

Comment:
Take this with a grain of salt, since I'm the newest to JavaScript, but I am finding lines 19 to 68 difficult to understand. Probably because of all the promise machinery. I wonder how much work it would take to rework this into a more declarative style using `async`/`await`, like the following:

```javascript
async componentDidMount() {
  try {
    const data = await instance.getBaseWithHttpInfo()
    const contains = [].concat(data.response.body.contains)
    this.setState({groupData: contains})
  } catch(error) {
    this.setState({message: error})
  }

  const groupData = this.state.groupData.map(async group => {
    const name = this.resourceToName(group)
    await instance.getGroupWithHttpInfo(name)
  })

  groupData.forEach(data => {
    try {
      let joined = []
      const groupName = this.resourceToName(data.response.body['@id'])
      data.response.body.contains.map(async c => {
        const name = this.resourceToName(c)
        const response_and_data = await instance.getResourceWithHttpInfo(groupName, name, { acceptEncoding: 'application/json' })
        joined.push({name: name, uri: c, id: response_and_data.response.body.id, group: 'ld4p'})
        this.setState({templatesForGroup: joined})
      })
    } catch(error) {
      this.setState({message: error})
    }
  })
}
```

I'm not sure this is exactly right; it's probably off by a fair bit, actually. But I put it forward as a comparison re: readability.

Again, grain of salt!

Reply:
@mjgiarlo I tried a little of what you were getting at, but this ends up being a promise that needs to be resolved:
```
const groupData = this.state.groupData.map(async group => {
      const name = this.resourceToName(group)
      await instance.getGroupWithHttpInfo(name)
    })
```

If you want to pair on Monday and see if we can work it out further, i'd be down for that!

Reply:
Sorry not to drop this message earlier but LDCX intervened. I have time this afternoon and tomorrow, if still helpful to chat.

Reply:
Thanks @mjgiarlo I hacked on this a little more, and it seems to me while it is possible to make the code more readable, it becomes much less testable. I got testability out of making those promises messages that could be passed into the parsing functions. That way I could mock the same promise/message as: 

`const promise = Promise.resolve(mockResponse(200, null, bodyContains))`

Is readability more important than testability? Maybe I'm making a false distinction because I'm just not familiar with how to mock and test a const in `componentDidMount()` ¯\_(ツ)_/¯

Reply:
Naw, that's debateable, and like I said... grain of salt! I am not blocking this PR, given my lack of familiarity with React and Javascript software design norms.

Reply:
Re: testability, my druthers are usually to create small methods/functions and then to mock those out as needed in tests. Since what you have already works though... :slightly_smiling_face: